### PR TITLE
add tenantId to driver metrics

### LINKED ISSUE #795: Add tenant tags to driver metrics
If possible, add tenant tags to the Cassandra driver metrics.

### PR DESCRIPTION
**Pull request by Yuqi-Du**
Adding tenantId to driver metrics, by implementing our own MetricIdGenerator.

**Which issue(s) this PR fixes**:
Fixes #795

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by Yuqi-Du** [comment]
big big thanks to Kathir's help. 
@kathirsvn

**Comment by jeffreyscarpenter** [comment]
Looks good. Can you add tests?

File: src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CustomTaggingMetricIdGenerator.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * A {@link MetricIdGenerator} that generates metric identifiers using a combination of names and
+ * tags.
+ *
+ * <p>Session metric identifiers contain a name starting with "session." and ending with the metric
+ * path, and a tag with the key "session" and the value of the current session name.
+ *
+ * <p>Node metric identifiers contain a name starting with "nodes." and ending with the metric path,
+ * and two tags: one with the key "session" and the value of the current session name, the other
+ * with the key "node" and the value of the current node endpoint.
+ */
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CustomTaggingMetricIdGenerator.java:40]: Thanks!

**Comment by Yuqi-Du** [comment]
> Looks good. Can you add tests?

added, thank

**Review by kathirsvn — APPROVED** [review]
LGTM

**Pull request merged**